### ISSUE TITLE
Popover interaction inconsistency

### ISSUE THREAD

**Issue by seanforyou23** [patternfly/patternfly-react#4210]
**Describe the issue. What is the expected and unexpected behavior?**

Popover doesn't forward focus correctly to close btn unless you use the enter key to launch it. 

VoiceOver (VO) screen readers use `cmd+option+space` as the primary way of selecting a thing. It's not uncommon to also use the enter key, but VO instructs the former when you focus on a clickable element. What I see is that a blind user wouldn't know the Popover was correctly launched when they use the `cmd+option+space` bar to launch it. The popover content is, by design, outside of the normal page content flow. Because the popovers are not live regions, nothing is announced other than the fact that the button is now "expanded" which is just a change in state of the button - not an indication or announcement of the new dynamic content. When new dynamic content appears on the screen, we generally want to send focus to it

In this state, the user hears the popover is expanded so they know its been launched, but unable to navigate to the content except through the rotor menu. This means they have to "figure out" that they need to press enter to get it to forward the VO cursor into the popover so that they can read the content like normal, or switch to another method of navigation to find the new content.

`cmd+options+space` fires a _click_ event on the focused element. If we can listen for that also, it could bring further alignment between the sighted and accessible experiences. 

**Please provide the steps to reproduce. Feel free to link CodeSandbox or another tool.**
Click a [Popover host element](https://www.patternfly.org/v4/documentation/react/components/popover#basic), notice it doesn't send focus to the Popover close button. Then, keyboard focus that same element, and press enter on your keyboard. Notice the focus is sent to the dynamic content.

<img width="454" alt="Screen Shot 2020-05-07 at 10 54 41 AM" src="https://user-images.githubusercontent.com/5942899/81309680-35590a80-9051-11ea-9b26-8efeb6fe00a4.png">

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?** Considering we generally want alignment between sighted and unsighted experiences, I'm inclined to call this a bug.

**What is your product and what release version are you targeting?** Ongoing PF development

Ping @jessiehuff @mcarrano
